src/NBitcoin/ChainBase.cs

```diff
@@ -4,15 +4,33 @@
 
 namespace NBitcoin
 {
+    /// <summary>Base implementation for chains.</summary>
     public abstract class ChainBase
     {
+        /// <summary>
+        /// Gets the block given a block id(hash).
+        /// </summary>
+        /// <param name="id">Block id to retreive.</param>
+        /// <returns>The block.</returns>
         public abstract ChainedBlock GetBlock(uint256 id);
+
+        /// <summary>
+        /// Gets the block at a given block height.
+        /// </summary>
+        /// <param name="height">Height to retreive block at.</param>
+        /// <returns>The block.</returns>
         public abstract ChainedBlock GetBlock(int height);
 
+        /// <summary>Gets the block at the tip of the chain.</summary>
         public abstract ChainedBlock Tip { get; }
 
+        /// <summary>Gets the height of the chain.</summary>
         public abstract int Height { get; }
 
+        /// <summary>
+        /// Enumerates from start of the chain.
+        /// </summary>
+        /// <returns>An enumerable iterator.</returns>
         protected abstract IEnumerable<ChainedBlock> EnumerateFromStart();
```

Comment:
this is not for this PR, just an idea - could we get rid of this method completely? just an area to explore

Reply:
Agreed. I find the ToEnumerable(fromTip) method confusing as well. I think it might be better to break into two separate methods rather than pass in a boolean.

Reply:
It might also be possible to implement the IEnumerator on chained block itself, but only if it will be really beneficial.